### DISCUSSION TITLE
Contributing to Outreachy - Dec 2022

### DISCUSSION THREAD

**Discussion by pgleeson** [ModECI/MDF#346] [General]
Hi!

Welcome to the Outreachy contribution period for Dec 2022. Thanks for your interest in MDF!

Please have a look at our [contribution guide](https://github.com/ModECI/MDF/blob/main/CONTRIBUTING.md) for how to get MDF installed on your local machine, and look at the [documentation](https://mdf.readthedocs.io/en/latest/) for the language. Feel free to open pull requests on those pages if anything is unclear and you have a suggestion for an update. 

I've created 2 suggested tasks and highlighted an issue which interested parties can look into over this period:

[Task 1- OSBv2 MDF notebook](https://github.com/ModECI/MDF/discussions/343)
[Task 2 - Keras to MDF](https://github.com/ModECI/MDF/discussions/345)
[Issue - Support PyTorch + Windows on py3.10](https://github.com/ModECI/MDF/issues/316)

Please note that [this project](https://www.outreachy.org/apply/project-selection/#modeci) will require some good previous experience in Python and ML. There are fewer open issues which are "low hanging fruit", mainly due to a very successful Outreachy Summer 2022! 

The one intern we are able to support this time will demonstrate that they have quickly developed an understanding of the MDF language and source code, and that they have a lot to contribute to (and gain from) the ModECI initiative for linking machine learning and neuroscience.

**Comment by Monsurat-Onabajo** [comment]
Thank you, i am super excited to work on this project and i am ready for all the challenges it will bring

**Comment by kusanele** [comment]
Thank you. Looking forward contributing to this project. I hope to learn a lot through this journey.

**Comment by Chigoziee** [comment]
Thanks, really excited to be working on this

**Reply by kusanele** [reply]
Welcome, looking forward to learning from you

**Comment by risper25** [comment]
Thanks, looking forward to working on this project.

**Comment by SoliatOmolara** [comment]
Thanks for this, looking forward to contributing to this project

**Comment by ting96haha** [comment]
Thank you for providing this exciting project! I am looking forward to contributing to this project to enhance my ML skills

**Reply by ting96haha** [reply]
Nice to meet you and thank you for providing this exciting project! @pgleeson

**Comment by kusanele** [comment]
@pgleeson , is there a way to use keras without TensorFlow? i just realized that my windows machine can't run TensorFlow since its a 32 bit system. 😑

**Reply by sanjayankur31** [reply]
I don't think so. Keras is built on top of Tensor Flow, so Tensor Flow is a mandatory dependency there :(

You can perhaps do some of the bits on the Open Source Brain v2 platform where both Keras and Tensor Flow should already be installed, (but that's probably not ideal for core development tasks)

**Reply by kusanele** [reply]
Thank you for your prompt response, i will to work on open Source Brain v2 for now , then will probably get a new system soon.

**Reply by pgleeson** [reply]
Yes, I think you can't really install keras without Tensorflow... 

FYI @kusanele if you do want to install TF on OSB you may run into this issue with large Python packages: https://github.com/OpenSourceBrain/OSBv2/issues/298. To install it use `pip install tensorflow  --no-cache-dir`, or even better `pip install tensorflow-cpu`, which just installs it for use on CPUs (not GPUs) as that's what's available on OSBv2.

**Reply by kusanele** [reply]
Thank you, Almost done with my second project , i will email the link soon

**Comment by Yusrah23** [comment]
Hello @pgleeson @sanjayankur31 I've completed a task for my contributions, here is the link: [Workspace](https://www.v2.opensourcebrain.org/workspace/open/421/jupyter)

**Comment by Monsurat-Onabajo** [comment]
Hello, good day
 i have worked on task one, am i to go ahead and work on the next task or to wait till i get a feedback on the task i worked on?

**Reply by pgleeson** [reply]
@Monsurat-Onabajo Yes, please go ahead and work on task 2 if you like.

**Comment by risper25** [comment]
Hello, I am also done too with task 1 . I have emailed my workspace link. Thanks.

**Comment by AniketARS** [comment]
Hello @sanjayankur31 @pgleeson I'm very happy to share that I've complete both the task (Task 1 and Task 2). I've learned many new thing in this journey specially loved the concept of making Graph from models.
Workspace link (JupyterLab): https://www.v2.opensourcebrain.org/workspace/open/427/jupyter
Workspace link: https://www.v2.opensourcebrain.org/workspace/427

**Reply by pgleeson** [reply]
Thanks @AniketARS. Nice work!

**Comment by Yusrah23** [comment]
Hello, @pgleeson @sanjayankur31 I need some clarification, please. Is it the link to the open-source workspace that I should add when I am recording my contribution on outreachy.

**Reply by sanjayankur31** [reply]
Yes, that'll be fine.It just needs to be something we can access :)

**Comment by pgleeson** [comment]
Regarding a question about the timeline which you should submit as part of the application... 

The timeline is for the applicant to define what tasks they will accomplish during the project, broken down into approx 2 week intervals. We won't specify exactly what should be put here, rather would be good to see what each applicant comes up with here themselves, based on their understanding of what needs to be done in the project. 

Doesn't have to be super detailed, it will just give us an impression of how you plan to manage your time, and what steps you think are important to achieve the goals of the project. The exact tasks to carry out will be refined over the period of the intership.

**Comment by Yusrah23** [comment]
Do you have any questions for us, in our final application because there is a space for the community-specific question that should be asked by our mentors @pgleeson @sanjayankur31 .

**Reply by pgleeson** [reply]
No community specific questions in the application. The important thing to get across in your final application though is that you have developed an understanding of the MDF language and the goals of the wider ModECI initiative and that you explain how you can make a significant contribution to this during the internship.
